learning/prompts/documentation-lookup/31_advanced_batched_dofn.md

```diff
@@ -0,0 +1,35 @@
+Prompt:
+What is a Batched DoFn in Apache Beam?
+Response:
+In Apache Beam, a [Batched DoFn](https://beam.apache.org/documentation/programming-guide/#batched-dofns) is a [DoFn](https://beam.apache.org/documentation/programming-guide/#pardo) that can operate on batches of logical elements. These `DoFns` leverage popular Python vectorized libraries like [NumPy](https://numpy.org/), [SciPy](https://scipy.org/) and [pandas](https://pandas.pydata.org/) to efficiently process batches of elements in a `PCollection` rather than processing each element individually.
```

Comment:
```suggestion
In Apache Beam, a [batched ‘DoFn’](https://beam.apache.org/documentation/programming-guide/#batched-dofns) is a specialized type of [‘DoFn’](https://beam.apache.org/documentation/programming-guide/#pardo) that operates on batches of logical elements within a [‘PCollection’](https://beam.apache.org/documentation/basics/#pcollection). Unlike regular ‘DoFn’s, which process each element individually, batched ‘DoFn’s enable the efficient processing of multiple elements together as batches.
```

Reply:
Suggested edits to explain what batched DoFns do in general first, then in the next paragraph we'll say that currently, they are only available in Python (and why).